From d6608d10d3a67c4cbfbfc5a27949f6b4aa10d1d8 Mon Sep 17 00:00:00 2001
From: Niklaus Xiao <stone_xy@live.cn>
Date: Tue, 16 May 2017 08:39:16 +0200
Subject: [PATCH 1119/1164] CDH-53943 HIVE-15483: Database and table name is
 case sensitive when used in show grant (Niklaus
 Xiao via Zoltan Haindrich)

Signed-off-by: Zoltan Haindrich <kirk@rxd.hu>
(cherry picked from commit 360a91e64590940641aff9304eb0275cbfd82e39)

Change-Id: Ica73d6d3c283e2d5d7ffa0dbc984b3bf7eb1ea67
---
 .../apache/hadoop/hive/metastore/ObjectStore.java  |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
index c8bfbb4..3897224 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
@@ -5514,6 +5514,8 @@ private void dropPartitionGrantsNoTxn(String dbName, String tableName, List<Stri
   public List<HiveObjectPrivilege> listTableGrantsAll(String dbName, String tableName) {
     boolean success = false;
     Query query = null;
+    dbName = HiveStringUtils.normalizeIdentifier(dbName);
+    tableName = HiveStringUtils.normalizeIdentifier(tableName);
     try {
       openTransaction();
       LOG.debug("Executing listTableGrantsAll");
@@ -5734,6 +5736,8 @@ private void dropPartitionGrantsNoTxn(String dbName, String tableName, List<Stri
       String columnName) {
     boolean success = false;
     Query query = null;
+    dbName = HiveStringUtils.normalizeIdentifier(dbName);
+    tableName = HiveStringUtils.normalizeIdentifier(tableName);
     try {
       openTransaction();
       LOG.debug("Executing listPrincipalTableColumnGrantsAll");
-- 
1.7.9.5

